Lesson45/gozde/recipe-app/src/example/Counter/__tests__/Counter.test.tsx

```diff
@@ -0,0 +1,86 @@
+import { render, screen } from "@testing-library/react";
+import { userEvent } from "@testing-library/user-event";
+import "@testing-library/jest-dom";
+import { Counter, counterReducer, CounterActionType } from "..";
+
+describe("<Counter /> component", () => {
+  it("should render correctly", () => {
+    render(<Counter />);
+
+    const countLabel = screen.getByTestId("count-label");
+    const countValue = screen.getByTestId("count-value");
+    const incrementButton = screen.getByTestId("increment-button");
+    const decrementButton = screen.getByTestId("decrement-button");
+
+    expect(countLabel).toBeInTheDocument();
+    expect(countValue).toBeInTheDocument();
+    expect(countValue).toHaveTextContent("0");
+    expect(incrementButton).toBeInTheDocument();
+    expect(decrementButton).toBeInTheDocument();
+  });
+
+  it("should render correctly with initial state specified", () => {
+    render(<Counter initialState={{ count: 5 }} />);
```

Comment:
This is a second test in this file that has a specific condition (count is set to 5). Since the test above already tested generic functionality (e.g. that elements rendered on the page correctly), it's usually ok to skip the same checks in this case. So keeping only:

```

const countValue = screen.getByTestId("count-value");
expect(countValue).toHaveTextContent("5");
```

would be sufficient. 
We do so to have less wordy tests and maintaining code takes effort.

No need to change anything now. :)